db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SelectNode.java

```diff
@@ -2853,4 +2911,62 @@ public JBitSet collectInnerTablesFromFlattenedOJ() {
         }
         return collected;
     }
+
+    private Map<Integer, List<ValueNode>> collectExpression() {
+        HashMap<Integer, List<ValueNode>> result = new HashMap<>();
+        boolean proceed = true;
```

Comment:
You're right. I was thinking that in case we find an expression that references more than one table, we know immediately that index on expressions defined on these tables cannot be covering. So we can quite the process earlier by not setting the `referencingExprssions` to these base tables. But this logic is buggy (for example, the whole map is cleared if we found one such expression, wrong for multiple tables' case anyway) and error-prone. So I removed it and always collect all expressions.